src/Components/AdvancedSearch/Internal/ColumnPicker/ColumnPicker.tsx

```diff
@@ -0,0 +1,122 @@
+import React from 'react';
+import {
+    IColumnPickerProps,
+    IColumnPickerStyleProps,
+    IColumnPickerStyles
+} from './ColumnPicker.types';
+import { getStyles } from './ColumnPicker.styles';
+import {
+    classNamesFunction,
+    useTheme,
+    styled,
+    Icon,
+    Dropdown,
+    IDropdownOption
+} from '@fluentui/react';
+import { useTranslation } from 'react-i18next';
+
+const getClassNames = classNamesFunction<
+    IColumnPickerStyleProps,
+    IColumnPickerStyles
+>();
+
+const ColumnPicker: React.FC<IColumnPickerProps> = ({
+    searchedProperties,
+    allAvailableProperties,
+    addColumn,
+    deleteColumn,
+    styles
+}) => {
+    const classNames = getClassNames(styles, {
+        theme: useTheme()
+    });
+    const { t } = useTranslation();
+    const [selectedKeys, setSelectedKeys] = React.useState<string[]>([]);
+    const [selectedPropsCount, setSelectedPropsCount] = React.useState(
+        searchedProperties.length
+    );
+
+    const onChange = (
+        event: React.FormEvent<HTMLDivElement>,
+        item: IDropdownOption
+    ): void => {
+        if (item) {
+            if (item.selected) {
+                setSelectedKeys([...selectedKeys, item.key as string]);
+                setSelectedPropsCount(selectedPropsCount + 1);
+                addColumn(item.key as string);
+            } else {
+                setSelectedKeys(selectedKeys.filter((key) => key !== item.key));
+                setSelectedPropsCount(selectedPropsCount - 1);
+                deleteColumn(item.key as string);
+            }
+        }
+    };
+
+    const onRenderTitle = (): JSX.Element => {
+        const placeholder = [
+            {
+                key: 'Icon',
+                text: `${t('advancedSearch.propertyCount', {
+                    selectedPropsCount
+                })}`,
+                data: { icon: 'ColumnOptions' }
+            }
+        ];
+        return (
+            <div className={classNames.dropdownTitle}>
+                {placeholder[0].data && placeholder[0].data.icon && (
```

Comment:
If you just defined placeholder above, this is always going to be true. You don't need this check